feat(cors): add Vercel URL for mobile camera testing (ALRAP-294)

MaksymChukhrai opened the pull request:
## Describe your changes
Added Vercel URL (`https://lab-ai-mobile-test.vercel.app`) to backend CORS whitelist for mobile yellow frame camera feature testing. This enables authentication and API calls from real Android/iOS devices during development.

## Issue ticket code
- [ALRAP-294](https://zenbittech.atlassian.net/browse/ALRAP-294)
- Parent: [ALRAP-211](https://zenbittech.atlassian.net/browse/ALRAP-211) - Doc Scanning through camera on Mobile Version

## Changes
- Updated `allowedOrigins` array in `src/main.ts`
- Added `https://lab-ai-mobile-test.vercel.app` for mobile testing
- Added inline comment explaining purpose (temporary testing URL)

## Why Vercel URL?
Mobile camera API requires HTTPS connection - `localhost` doesn't work on real devices. Created alternative frontend repository for rapid cross-device testing:
- Real iPhone/Android testing
- OAuth callbacks support
- Instant deployment on push

## Testing Flow
1. Deploy backend to Heroku
2. Test auth from mobile devices via Vercel URL
3. Implement ALRAP-211 camera feature
4. Merge tested code to main frontend repo

### General Checklist

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [x] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters (no new parameters added)
- [x] Imports should come in a specific order: node modules, absolute path, relative path
- [x] Don't have "any" on my code
- [x] Used the try/catch pattern for error handling (not applicable - static config)
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings (not applicable - URL whitelist)
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] No links in the code, env links should be in env file (CORS whitelist is exception - URLs expected here)
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants (not applicable)
- [x] Functions are public only if it's used outside the class (not applicable)
- [x] No hardcoded values (CORS whitelist URLs are expected to be hardcoded)
- [x] Covered by tests (not applicable - static configuration, no business logic)
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/)

### Backend Checklist

- [x] Swagger documentation updated (no API changes)
- [x] Database requests are optimized and not redundant (no DB changes)
- [x] Unit tests written (not applicable - static CORS config, no testable logic)
- [x] Use ConfigService instead of process.env (not applicable - static whitelist preferred over env)
- [x] Use transactions if there is a call chain that mutates data in different tables (not applicable - no DB operations)
- [x] Use @index decorator for frequently requested data (not applicable - no entity changes)

### Testing Checklist
- [ ] Will test after deploy to Heroku
- [ ] Verify CORS headers from Vercel URL
- [ ] Test OAuth login on mobile
- [ ] Confirm no errors in Heroku logs

## Related PRs
- Frontend PR: [will be created after backend deploy]

## Post-Merge Actions
1. Deploy to Heroku production
2. Test on mobile devices
3. Notify team in ALRAP-294 Jira comments
4. Proceed with ALRAP-211 implementation

MaksymChukhrai committed:
feat(cors): add Vercel URL for mobile camera testing

- Added https://lab-ai-mobile-test.vercel.app to CORS whitelist
- Required for ALRAP-294 mobile yellow frame feature testing
- Temporary URL for cross-device testing (iPhone/Android)

ALRAP-294

lyudad commented on src/main.ts:
```diff
@@ -49,6 +49,7 @@ async function bootstrap() {
   const allowedOrigins = [
     'http://localhost:5173',
     'https://green-frontend-bbac1a42fc92.herokuapp.com',
+    'https://lab-ai-mobile-test.vercel.app',
```
Все эти переменные закинь в енвы

MaksymChukhrai replied:
Fixed
Changes:

- Moved hardcoded CORS origins from main.ts to ALLOWED_ORIGINS env variable
- Added Joi validation with type safety for comma-separated URIs
- Updated .env.example with production configuration example
- All tests passing (10/10) 

Files Modified:

- src/main.ts - replaced hardcoded array with env parsing
- src/config/env.validation.ts - added ALLOWED_ORIGINS validation with type guards
- .env.example - added ALLOWED_ORIGINS with production example

MaksymChukhrai committed:
refactor(config): move CORS allowed origins to environment variables

- Added ALLOWED_ORIGINS env variable (comma-separated list)
- Updated env.validation.ts with type-safe Joi validator
- Refactored main.ts to parse origins from ConfigService
- Added validation for empty origins and invalid URIs
- Kept backward compatibility with FRONTEND_URL fallback
- Updated .env.example with production examples

ALRTAP-287

MaksymChukhrai committed:
minor correction

lyudad commented on src/config/env.validation.ts:
```diff
@@ -1,21 +1,21 @@
+  PORT: Joi.number().default(3000),
 
-  DB_TYPE: Joi.string().valid('mysql').default('mysql').required(),
+  DB_TYPE: Joi.string().valid('mysql').required(),
```
Why you change the original line?

MaksymChukhrai replied:
Fixed

lyudad commented on src/config/env.validation.ts:
```diff
@@ -1,21 +1,21 @@
   NODE_ENV: Joi.string()
     .valid('development', 'production', 'test')
     .default('development'),
+  PORT: Joi.number().default(3000),
```
Why you move it?

MaksymChukhrai committed:
refactor(cors): add ALLOWED_ORIGINS validation only

- Added ALLOWED_ORIGINS env variable validation with type guards
- No reordering of existing variables
- No comments (as per code style)

MaksymChukhrai commented:
@lyudad Fixed! ✅

**Changes made:**
- ✅ Reverted reordering of `PORT` and `JWT_REFRESH_SECRET`
- ✅ Added only `ALLOWED_ORIGINS` validation block
- ✅ Tests passing (10/10)

**Diff summary:** +38 lines (only new `ALLOWED_ORIGINS` validation)

Thank you for the review!
